forecasting/src/autogluon/forecasting/models/presets.py

```diff
@@ -0,0 +1,199 @@
+            },
+            "MQCNN": {"epochs": 5, "num_batches_per_epoch": 10, "context_length": 5},
+            "DeepAR": {"epochs": 5, "num_batches_per_epoch": 10, "context_length": 5},
+        },
+        "toy_hpo": {
+            "SimpleFeedForward": {
+                "epochs": 5,
+                "num_batches_per_epoch": 10,
+                "context_length": ag.Int(5, 25),
+            },
+            "MQCNN": {
+                "epochs": 5,
+                "num_batches_per_epoch": 10,
+                "context_length": ag.Int(5, 25),
+            },
+            "DeepAR": {
+                "epochs": 5,
+                "num_batches_per_epoch": 10,
+                "context_length": ag.Int(5, 25),
+            },
+        },
+        "default": {
+            "SimpleFeedForward": {},
+            "MQCNN": {},
+            "DeepAR": {},
+            # "AutoTabular": {} # AutoTabular model is experimental.
+        },
+        "default_hpo": {
+            "MQCNN": {
+                "context_length": ag.Int(
+                    min(prediction_length, max(10, 2 * prediction_length), 250),
+                    max(min(500, 12 * prediction_length), 4 * prediction_length),
+                    default=prediction_length * 4,
+                ),
+            },
+            "DeepAR": {
+                "context_length": ag.Int(
+                    min(prediction_length, max(10, 2 * prediction_length), 250),
+                    max(min(500, 12 * prediction_length), prediction_length),
+                    default=prediction_length,
+                ),
+            },
+            "SimpleFeedForward": {
+                "context_length": ag.Int(
+                    min(prediction_length, max(10, 2 * prediction_length), 250),
+                    max(min(500, 12 * prediction_length), prediction_length),
+                    default=prediction_length,
+                ),
+            },
+        },
+    }
+    return default_model_hps[key]
+
+
+def get_preset_models(
+    freq: str,
+    prediction_length: int,
+    path: str,
+    eval_metric: str,
+    hyperparameters: Union[str, Dict],
+    hyperparameter_tune: bool,
+    use_feat_static_cat: bool,
+    use_feat_static_real: bool,
+    cardinality: int,
+    **kwargs,
+):
+    """
+    Create a list of models according to hyperparameters. If hyperparamaters=None,
+    will create models according to presets.
+    """
+    models = []
+    if isinstance(hyperparameters, str):
+        hyperparameters = copy.deepcopy(
+            get_default_hps(hyperparameters, prediction_length)
+        )
+    else:
+        hp_str = "default" if not hyperparameter_tune else "default_hpo"
+        default_hps = copy.deepcopy(get_default_hps(hp_str, prediction_length))
+
+        if hyperparameters is not None:
+            # filter only default_hps for models with hyperparameters provided
+            default_hps = {
+                model: default_hps.get(model, {}) for model in hyperparameters
```

Comment:
Nitpick: would `hp` be a better name when iterating through hyperparameters? A `hp` can be used to create a model.

Reply:
No these are model names indeed. so the hyperparameters dict maps model names to a nested dict of hyperparameters.